src/base/multithreading.cc

```diff
@@ -32,6 +33,24 @@ static const std::thread::id kNoThread;
 static std::thread::id initializer_thread(kNoThread);
 static std::thread::id logic_thread(kNoThread);
 
+// To protect the global mutex list
+std::mutex MutexLock::s_mutex_;
+static std::mutex responsiveness_list_mutex_;
+
+std::vector<std::function<void()>> MutexLock::stay_responsive_;
+static std::vector<std::pair<std::thread::id, std::thread::id>> acting_as_another_thread;
+
+/** Wrapper around a STL recursive mutex plus some metadata. */
+struct MutexRecord {
+	std::recursive_mutex mutex;  ///< The actual mutex.
+	std::set<std::thread::id>
```

Comment:
could this have a std:string name?

Reply:
That's what `to_string(MutexLock::ID)` is for. Putting this in the mutex record would be fragile since then it would have to be initialized somewhere, and there's no good place for that.